rfcs/2021-19-02-units-for-typography-and-layout.md

```diff
@@ -0,0 +1,603 @@
+
+Let's see how things look on devices with different resolution and densities with largest device font size
+
+
+<img alt="Layout and typography at 200% zoom" src="./images/unit-largest-font-size-app.png" width="1400px">
+
+## Absolute unit vs relative unit?
+Until now we saw how typography and layout reacts to font sizes and zoom with different units. Referring to [our checklist](#checklist) gives us 3 options:
+1. **Everything `px`(web)**
+   1. Changing default font-size on the browser level doesn't changes the text content size.
+   2. Changing default font-size on the browser level doesn't changes the layout sizes. Nothing breaks as everything is in absolute unit.
+   3. Page zoom works as expected and the layout never breaks.
+   4. Image when font-size is `32px`
+   <img alt="Layout px text px" src="./images/unit-layout-px-text-px.png" width="300px">
+2. **Everything `rems`(web)**
+   1. Changing default font-size changes the font size of the content.
+   2. Changing default font-size changes layout sizes relatively. Things mostly doesn't break as everything is in ratios.
+   3. Page zoom works as expected and the layout mostly doesn't breaks.
+   4. Image when font-size is `32px`
+   <img alt="Layout rem text rem" src="./images/unit-layout-rem-text-rem.png" width="300px">
+3. **Typography `rems` and Layout `px`(web)**
+   1. Changing default font-size changes the font size of the content.
+   2. Changing default font-size changes doesn't changes the layout sizes. Some layout breakages can happen as the text may end up being bigger than you expect in your layout.
+   3. Page zoom works as expected.
+   4. Image when font-size is `32px`
+   <img alt="Layout px text rem" src="./images/unit-layout-px-text-rem.png" width="300px">
+4. **Everything `dp`(native)**
+   1. Changing device font size changes the font size of the content.
+   2. Changing device font size doesn't changes the layout size.
+   3. Image when device font-size is largest `@1x`
+   <img alt="Layout dp 1x" src="./images/unit-layout-dp-1x.png" width="300px">
+
+   4. Image when device font-size is largest `@3x`
+   <img alt="Layout dp 3x" src="./images/unit-layout-dp-3x.png" width="300px">
+5. **Tyography auto scale, layout pixel ratio(native)**
+   1. Changing device font size changes the font size of the content.
+   2. Changing device font size doesn't changes the layout size. Some layout breakages can happen as the text may end up being bigger than you expect in your layout.
+   3. Things just look bigger if we use pixel ratio as a multiplicative factor as react native uses `dp` it automatically does those things for us.
+   4. Image when device font-size is largest `@1x`
+   <img alt="Layout pr 1x" src="./images/unit-layout-pr-1x.png" width="300px">
+
+   5. Image when device font-size is largest `@3x`
+   <img alt="Layout pr 3x" src="./images/unit-layout-pr-3x.png" width="300px">
+### Absolute unit vs relative unit matrix
```

Comment:
```suggestion
### Absolute unit vs relative unit matrix (on changing base font size)
```

Or something better to clarify what the "changes" are for in the following table.

Reply:
umm, it's not just because of change in base font size this matrix also dictates about page zoom.